sky/adaptors/aws.py

```diff
@@ -3,9 +3,25 @@
 # pylint: disable=import-outside-toplevel
 
 import functools
+import threading
 
 boto3 = None
 botocore = None
+func_lock = threading.RLock()
```

Comment:
nit: rename to `_session_creation_lock = ..`?

Reply:
Good point! Fixed. Thanks!